portal/app/core/assets/js/weather.js

```diff
@@ -2,24 +2,42 @@ var weather = weather || {};
 
 $(document).ready(function(ns) {
     var onLoad = function() {
-        // setup listeners
-        setupListeners();
-    },
+            // setup listeners
+            setupListeners();
+        },
 
-    setupListeners = function() {
-        $("#get-weather").on("click", function() {
-            ns.queryWeatherBasedOnZip($("#get-weather-inpt").val());
-        });
-    };
+        setupListeners = function() {
+            $("#get-weather").on("click", function() {
+                ns.queryWeatherBasedOnZip($("#get-weather-inpt").val());
+            });
+        };
 
     ns.queryWeatherBasedOnZip = function(zipCode) {
         // Query the OWM API for weather data
+        data = JSON.stringify(data, null, '\t');
         var url = "https://api.openweathermap.org/data/2.5/weather?zip=" + zipCode +
-                  ",us&APPID=6faafe3de999509b86b118803ee2ca8f";
+            ",us&APPID=6faafe3de999509b86b118803ee2ca8f";
         $.get(url, function(data) {
-            $("#weather-response").html(JSON.stringify(data, null, '\t'));
+            $("#weather-response").html(data);
         });
     };
 
     onLoad();
-}(weather));
\ No newline at end of file
+}(weather));
+
+
+// weather functionality for all pages
+navigator.geolocation.getCurrentPosition(function(position) {
+    latitude = position.coords.latitude;
+    longitude = position.coords.longitude;
+    var urlPosition = "https://api.openweathermap.org/data/2.5/weather?lat=" +
+        latitude + "&lon=" + longitude + "&APPID=6faafe3de999509b86b118803ee2ca8f";
```

Comment:
You're still not DRY enough. Why are you repeating the APPID and re-specifying the url?

There should be one single function that calls open weather map api

Reply:
Let me know what you think.